### PR TITLE
refactor handling of theme

### PR DESCRIPTION
**Pull request by freeman-lab**
This breaking change refactors the package in two key ways
- changes the handling of themes to remove a dependency on `theme-ui`
- removes the `Colorbar` component, which we have decided to move to our `@colormaps/components` repository.

Most of the core colormap logic now lives within `makeColormap` (previously `generator`) which only depends on `chroma.js`. That can be used outside of react and without hooks. We include as a helper function a `useColormap` but it is a very thin layer around `makeColormap` that wraps the generation step inside a `useMemo`, and otherwise has the exact same arguments as `makeColormap`.

This means that any consumer wanting to use themes via `theme-ui` must do:

```
import { useColorMode } from 'theme-ui'
import { useColormap } from '@carbonplan/colormaps'

const [mode, ] = useColorMode()
colors = useColormap('reds', {mode: mode})
```

While the new version is a bit more verbose for the consumer, we create the flexibility to incorporate into other patterns for specifying themes (or not using themes), which seems like a net benefit. 

If we wanted to, we could create a `useThemedColormap` that wrapped the above four lines to give us

```
import { useThemedColormap } from '@carbonplan/...'

colors = useThemedColormap('reds')
```

Is that worth doing? And where would it live?

### CODE REVIEW

**Comment by freeman-lab** [comment]
Also, I've "tested" this by pulling it into our design site (with a small change to handle the modes), and I confirmed exact visual parity and proper handling of modes.

**Comment by freeman-lab** [comment]
I went ahead and added a `useThemedColormap`. Because the dependency on `theme-ui` should be able to be pruned away, this feels like a nice convenience without creating any barriers for those not using `theme-ui`, but happy to reconsider.

File: src/use-themed-colormap.mjs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { useColorMode } from 'theme-ui'
+import useColormap from './use-colormap'
+
+const useThemedColormap = (name, options) => {
+	const [mode, ] = useColorMode()
```
Comment (katamartin) [inline src/use-themed-colormap.mjs:5]: Hm I wonder why this isn't getting picked up by the linter? 🤔 

The code here looks good though!

File: src/use-colormap.mjs [hunk 2]
```diff
@@ -1,27 +1,10 @@
...
-    throw Error(`requested colormap '${name}' is not defined`)
-  }
+import makeColormap from './make-colormap'
 
+const useColormap = (name, options) => {
   const colormap = useMemo(() => {
-    return generator(name, colorMode, count, format)
-  }, [name, colorMode, count])
+    return makeColormap(name, options)
+  }, [name, options.count, options.format, options.mode])
```
Comment (katamartin) [inline src/use-colormap.mjs:7]: Hm it seems like these checks are the only thing preventing this hook from being used like `useColormap('warm')`, maybe we could do:
```suggestion
  }, [name, options?.count, options?.format, options?.mode])
```

**Review by katamartin — COMMENTED** [review]
Left a couple of small concerns, but the changes here--including the `useThemedColormap` hook--largely look good to me!

File: src/use-colormap.mjs (hunk 2 above)
Comment (freeman-lab) [reply src/use-colormap.mjs]: fixed

File: src/use-themed-colormap.mjs (hunk 1 above)
Comment (freeman-lab) [reply src/use-themed-colormap.mjs]: fixed!

**Review by katamartin — APPROVED** [review]
lgtm!

**Pull request merged**